relayer/relayer.go

```diff
@@ -284,6 +284,62 @@ func (r *Relayer) ReconnectToSubscriber() error {
 	return nil
 }
 
+// Fetch the appropriate message relayer
+// Checks for the following registered keys. Exactly one of these keys should be registered, or none of them.
+// 1. An exact match on sourceBlockchainID, destinationBlockchainID, originSenderAddress, and destinationAddress
+// 2. A match on sourceBlockchainID and destinationBlockchainID, with a specific originSenderAddress and any destinationAddress
+// 3. A match on sourceBlockchainID and destinationBlockchainID, with any originSenderAddress and a specific destinationAddress
+// 4. A match on sourceBlockchainID and destinationBlockchainID, with any originSenderAddress and any destinationAddress
+func (r *Relayer) getMessageRelayer(
```

Comment:
instead of checking for all 4 cases, can something with prior knoweldge of the source blockchain's supported destinations, save information about which of the 4 cases each supported destination is?

Reply:
We could probably optimize this function by implementing a more intelligent matcher, but that seems like premature optimization. We could also add a cache of registered relayer IDs that we could check instead of re-computing them each time, but that also seems premature. This function isn't exactly expensive when considering that message relaying is heavily network-bound.